### DISCUSSION TITLE
Continuous build ef2b619

### DISCUSSION THREAD

**Discussion by Terracotta-6** [artisan-roaster-scope/artisan#1366] [General]
The time axis of foreground Profiles is different from that of background Profiles.
The Simulator  The time axis of offset.
![68be52ae5104d79bb4a744b95c6a08b](https://github.com/artisan-roaster-scope/artisan/assets/149906183/36a10f8c-7350-4340-a377-e46864d814e9)
![2aced421c844e9725a885effc0b3158](https://github.com/artisan-roaster-scope/artisan/assets/149906183/c3c00dc1-c4c2-45f7-ae03-bcb41dfd3cb8)

**Comment by Terracotta-6** [comment]
[Simulator & Switch Profiles test_23-12-25_0642.alog.txt](https://github.com/artisan-roaster-scope/artisan/files/13782464/Simulator.Switch.Profiles.test_23-12-25_0642.alog.txt)
[Uploading artisan-settings.aset.txt…]()

https://github.com/artisan-roaster-scope/artisan/assets/149906183/8afc8d8c-7619-47df-a692-9ef1fe11a785

**Reply by MAKOMO** [reply]
Thanks for reporting this one. I found and fixed it with Commit cfa6dad9882e50ca7024d428e1456edaa900356e

**Reply by Terracotta-6** [reply]
Foreground and background curves perfectly match. Thank you.

**Comment by roasterdave** [comment]
Unfortunately the settings file did not attach so I can't see your true settings.  Hopefully the following correctly explains what is happening on your machine.

Top screenshot - it looks like you may have hit the right arrow key.  That will shift the background profile some number of seconds as defined in Roast>> Background.  When the background is shifted the event marks and their times are not changed. 

![bkg2](https://github.com/artisan-roaster-scope/artisan/assets/30009312/4da9fadf-a0c0-4adf-9cc5-2ac2c45b8a3f)

Bottom screenshot and video - it looks like you selected Roast>> Background **Align CHARGE** to align background and foreground curves on the CHARGE event.  In the simulation you manually hit CHARGE 25 seconds after START.  Thus the background profile was shifted 25 seconds to the right to align its CHARGE event with the recording's CHARGE event.

![bkg3](https://github.com/artisan-roaster-scope/artisan/assets/30009312/d038dc48-d704-4a68-b0e1-0b09e7b3dfd6)

**Reply by Terracotta-6** [reply]
https://github.com/artisan-roaster-scope/artisan/assets/149906183/99959bdd-c283-4fd5-a023-c2624b75a9a7

**Reply by Terracotta-6** [reply]
https://github.com/artisan-roaster-scope/artisan/assets/149906183/42dc23ec-0078-4419-8df3-34b6f1eda750

**Reply by Terracotta-6** [reply]
https://github.com/artisan-roaster-scope/artisan/assets/149906183/9ece7038-b57b-491f-928c-f6e6f1500289

**Comment by roasterdave** [comment]
I see the shift now. We will investigate.  Thanks for reporting.

**Reply by Terracotta-6** [reply]
Have been fixes. Thank you .

**Comment by Terracotta-6** [comment]
Curves Rate of Rise Filter.

https://github.com/artisan-roaster-scope/artisan/assets/149906183/214f84a2-80fa-44a1-9d7b-c5e58da168bf

**Reply by Terracotta-6** [reply]
[CTE.Silon.Touch.2023-12-28.alog.txt.txt](https://github.com/artisan-roaster-scope/artisan/files/13851747/CTE.Silon.Touch.2023-12-28.alog.txt.txt)

**Reply by roasterdave** [reply]
The settings file did not upload.  Please re-post.  
Also, please post this alog file.
Thank you.

**Comment by Terracotta-6** [comment]


**Reply by roasterdave** [reply]
A link back to this discussion is inserted instead of an uploaded file.  Can you please try again?

**Reply by Terracotta-6** [reply]
Reupload as you requested.

**Reply by roasterdave** [reply]
Got them now.  Thanks

**Comment by Terracotta-6** [comment]
Profile Designer-  The total save time of curves after editing is lost by 1 second.
![e54137a2285ed885a03c37a960d24db](https://github.com/artisan-roaster-scope/artisan/assets/149906183/05eed80e-ace9-4f42-9689-ebb615fea6a5)

![e27c547757593e8d4f58ee9eaf0a4ce](https://github.com/artisan-roaster-scope/artisan/assets/149906183/1cc78c27-646c-4436-8de4-132397be067f)


https://github.com/artisan-roaster-scope/artisan/assets/149906183/71bd3f08-492f-4811-815e-8ca515ea02da

[Uploading 24-01-07_1659.alog,txt…]()

**Reply by roasterdave** [reply]
This alog file did not upload.  Would you please re-upload it?  Thanks.

**Reply by Terracotta-6** [reply]
[24-01-07_1659.zip](https://github.com/artisan-roaster-scope/artisan/files/13855149/24-01-07_1659.zip)

**Reply by Terracotta-6** [reply]
Have been fixes. Thank you .
